File: Python/RotateArrayTest.py
from unittest import TestCase

from RotateArray import RotateArray


class TestRotateArray(TestCase):
    def test_rotate(self):
        ra = RotateArray()

        array0 = [1]
        ra.rotate(array0, 1)
        self.assertEqual(array0, [1])

        array1 = [1, 2]
        ra.rotate(array1, 1)
        self.assertEqual(array1, [2, 1])

        array2 = [1, 2, 3, 4, 5, 6, 7]
        ra.rotate(array2, 3)
        self.assertEqual(array2, [5, 6, 7, 1, 2, 3, 4])

File: Golang/RotateArray/RotateArray_test.go
package RotateArray

import (
	"testing"
	"reflect"
)

func TestRotate(t *testing.T) {
	array0 := []int{1}
	rotate(array0, 1)
	if reflect.DeepEqual(array0, []int{1}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	array1 := []int{1, 2}
	rotate(array1, 1)
	if reflect.DeepEqual(array1, []int{2, 1}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	array2 := []int{1, 2, 3, 4, 5, 6, 7}
	rotate(array2, 3)
	if reflect.DeepEqual(array2, []int{5, 6, 7, 1, 2, 3, 4}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

}
